apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,242 @@
+  getAllBuildVariants,
+  getAllTaskStatuses,
+  groupTestsByName,
+} from "./utils";
+
+const { green } = palette;
+interface TestAnalysisProps {
+  versionId: string;
+}
+const TestAnalysis: React.FC<TestAnalysisProps> = ({ versionId }) => {
+  const [selectedTaskStatuses, setSelectedTaskStatuses] = useQueryParam<
+    string[]
+  >(TestAnalysisQueryParams.Statuses, []);
+
+  const [selectedBuildVariants, setSelectedBuildVariants] = useQueryParam<
+    string[]
+  >(TestAnalysisQueryParams.Variants, []);
+
+  const [testName, setTestName] = useQueryParam(
+    TestAnalysisQueryParams.TestName,
+    "",
+  );
+
+  const dispatchToast = useToastContext();
+  const { data, loading } = useQuery<
+    TestAnalysisQuery,
+    TestAnalysisQueryVariables
+  >(TEST_ANALYSIS, {
+    variables: {
+      versionId,
+      options: {
+        statuses: failedTaskStatuses,
+      },
+      opts: {
+        statuses: ["fail"],
+      },
+    },
+    onError: (err) => {
+      dispatchToast.error(`Error fetching test analysis: ${err.message}`);
+    },
+  });
+
+  const groupedTestsMap = useMemo(
+    () => groupTestsByName(data ? data?.version?.tasks?.data : []),
+    [data],
+  );
+
+  const buildVariants = useMemo(
+    () => getAllBuildVariants(groupedTestsMap),
+    [groupedTestsMap],
+  );
+  const taskStatuses = useMemo(
+    () => getAllTaskStatuses(groupedTestsMap),
+    [groupedTestsMap],
+  );
+  let filteredGroupedTestsMap = new Map<string, TaskBuildVariantField[]>();
+  try {
+    filteredGroupedTestsMap = filterGroupedTests(
+      groupedTestsMap,
+      testName,
+      selectedTaskStatuses,
+      selectedBuildVariants,
+    );
+  } catch (error) {
+    reportError(new Error(`Invalid Regexp: ${error}`)).severe();
+    dispatchToast.error(`Invalid Regexp: ${error}`);
+  }
+
+  const groupedTestsMapEntries = Array.from(
+    filteredGroupedTestsMap.entries(),
+  ).sort((a, b) => b[1].length - a[1].length);
+
+  const numberOfTestsThatFailedOnMoreThanOneTask =
+    groupedTestsMapEntries.filter(([, tasks]) => tasks.length > 1).length;
+
+  const totalTestCount = countTotalTests(groupedTestsMap);
+  const totalFilteredTestCount = countTotalTests(filteredGroupedTestsMap);
+
+  const hasMatchingResults = totalFilteredTestCount > 0;
+  const hasResults = data && totalTestCount > 0;
+  return (
+    <Container>
+      {loading ? (
+        <ListSkeleton />
+      ) : (
+        <>
+          <Title>
+            {numberOfTestsThatFailedOnMoreThanOneTask}{" "}
+            {pluralize("test", numberOfTestsThatFailedOnMoreThanOneTask)} failed
+            across more than one task
+          </Title>
+          <Body>
+            This page provides an overview of all test failures in the current
+            version. If a test fails across multiple tasks, it could suggest a
+            flaky test or a broader issue.
+          </Body>
+          <FilterContainer>
+            <div>
+              <LabelWrapper>
+                <Label
```

Comment:
Could you add disabled styling to this? I also wouldn't be opposed to using the regular TextInput since it has more features built in and we're not using the search results components.
<img width="783" alt="image" src="https://github.com/user-attachments/assets/d51e3358-50fb-421b-8b0d-04c7c8f61945">

Reply:
TIL labels have disabled styling. This does use the regular TextInput under the hood with the addition of some validation logic. I switched it to using the default label prop instead of the random extra label. 🤦